core/primitives/src/shard_layout.rs

```diff
@@ -241,9 +241,34 @@ pub enum ShardLayoutError {
 }
 
 impl ShardLayout {
-    /* Some constructors */
-    pub fn v0_single_shard() -> Self {
-        Self::v0(1, 0)
+    /// Construct a layout with a single shard
+    pub fn single_shard() -> Self {
```

Comment:
The PR is not quite ready. But I'd love to start getting feedbacks for how to proceed.

This function isn't really necessary. Any reasons we should keep it?

Reply:
It looks quite convenient. I may take advantage of it when testing the non-contiguous ShardIds.